lib/bitstyles_phoenix/bitstyles.ex

```diff
@@ -129,24 +129,52 @@ defmodule BitstylesPhoenix.Bitstyles do
         _ -> class
       end
 
-    classname(mapping, "1.5.0")
+    classname(mapping, {1, 5, 0})
   end
 
-  def classname(_class, version) do
+  def classname(_class, version) when is_tuple(version) do
     raise("""
-    The version #{version} of bitstyles is not supported. The helpers will produce incorrect classes.
+    The version #{version_to_string(version)} of bitstyles is not supported. The helpers will produce incorrect classes.
     Please upgrade bitsyles and set the `bitsyles_version` to the updated version.
     """)
   end
 
-  def version do
+  def version(format \\ :tuple) do
```

Comment:
>  I think I would change all other places to tuples (like the showcases etc), which will simplify things here quite a lot.

It is not possible to avoid strings for technical reasons, for example in all those places it must be a string:

- `cdn_url`
- `IO.warn` for version not supported (too low or too high) in `classnames`
- in HTML as link text and headings etc., and for HTML file names in generate_version_showcase

I could change more places to use tuples (e.g. keyword keys in stories when a story has tests for multiple versions, a list of supported versions in generate_version_showcase) but I can't avoid having to convert them back to strings again

Reply:
> e.g. keyword keys in stories when a story has tests for multiple versions,

well, changing that would mean it's no longer a keyword list, but either a list of a map, which would mean I need to touch every story to change the last argument to the new data structure.

In my mind, I was trying to keep as much of the code unchanged, and only deal with tuples where necessary (for comparisons). I find writing versions in strings a bit nicer because it looks more like you expect version numbers to look in various places (git tags, in changelogs etc.)